homeassistant/components/switch/dlink.py

```diff
@@ -135,10 +135,34 @@ def __init__(self, smartplug):
         self.temperature = None
         self.current_consumption = None
         self.total_consumption = None
+        self.available = False
+        self._n_tried = 0
+        self._last_tried = None
 
     def update(self):
         """Get the latest data from the smart plug."""
-        self.state = self.smartplug.state
+        if self._last_tried is not None:
```

Comment:
Use trottle on a data object. After that you can remove this retry handling and keep it simple with same effect 👍 Yes, you can lost 10 seconds of data, but that is the problem of client library and not our problem.

Reply:
Thanks for the review.
I do not think that Trottle is optimal here.
Sometimes the switch becomes unresponsible. If HA then tries to reconnect to often, it seems that the switch get stuck forever and would require a hard reset. 

So I want to wait longer than the normal update frequency after a failure.

Reply:
It is true that the switch has anit-brute forcing code build in so you cannot query it too often. @Danielhiversen can you confirm that his fixes the issue? I tried some similar changes a while back and it did not really change anything.

Reply:
I have seen similar behavior when looking into the bug but let's hope this fixes it.